src/js/angular/import/directives/import-resource-tree.directive.js

```diff
@@ -195,7 +222,8 @@ function importResourceTreeDirective($timeout, ImportContextService) {
             const updateListedImportResources = () => {
                 $scope.resources.getRoot().updateSelectionState();
                 sortResources();
-                $scope.displayResources = $scope.resources.toList()
+                $scope.checkboxControlModel = new CheckboxControlModel($scope.resources.toList());
+                $scope.displayResources = $scope.checkboxControlModel.getResourcesList()
```

Comment:
The idea of this ne model/utility is to hide the internal implementation. In other words, there shouldn't have such filtering and other manipulations on the underlying array here. All this must happen in the wrapper. Otherwise it's pointless because the client would in fact know about the underlying structure and would be able to break it at some point.